VitalHealthConnect/src/main/java/io/tryvital/vitalhealthconnect/workers/ResourceSyncWorker.kt

```diff
@@ -165,14 +165,10 @@ internal class ResourceSyncWorker(appContext: Context, workerParams: WorkerParam
                 vitalLogger.logI("${input.resource}: skipped because backend status is ${backendState.status}")
                 return Result.success()
             }
-            UserSDKSyncStatus.Active -> when (state) {
-                // Prefer backend provided pull range if present.
-                is ResourceSyncState.Historical -> state.copy(
-                    start = backendState.requestStartDate ?: state.start,
-                    end = backendState.requestEndDate ?: state.end,
-                )
-                is ResourceSyncState.Incremental -> state
```

Comment:
`ResourceSyncState.Incremental` is not the same class as `ResourceSyncState.Historical`.

The former (as is) doesn't have concept of "end bound", only a "changes token" to fetch next batch of changes.

You will need to start with extending `ResourceSyncState.Incremental` with an optional field. Then make sure the `incrementalBackfill` process does honour the end bound.

Reply:
To enforce the end bound in Health Connect, it can be more complicated than Apple HealthKit.

All HealthKit queries can accept time-based filters/predicates, including the incremental AnchoredObjectQuery used for most samples, sleeps & workouts.

Meanwhile, HealthConnect's `getChanges()` incremental query doesn't... so we will need to apply the optional filtering step to `ChangesResponse` ourselves.